photutils/background/background_2d.py

```diff
@@ -745,8 +745,8 @@ def _calc_coordinates(self):
         self.yx = np.column_stack([self.y, self.x])
 
         # the position coordinates used when calling an interpolator
-        ny, nx = self.data.shape
-        self.data_coords = np.array(list(product(range(ny), range(nx))))
+        nx, ny = self.data.shape
+        self.data_coords = np.indices((ny, nx)).T.reshape(nx*ny, 2)
```

Comment:
```suggestion
        self.data_coords = np.indices((nx, ny)).T.reshape(nx*ny, 2)[:, [1, 0]]
```

While reviewing this PR, I discovered that both this change and the version in master were both incorrect because of the order of the coordinates.  I fixed the version in master in #1104 and added a simple regression test.  That's the currently failing test, and it's a real test failure.  The suggested changes above fix this issue (after you rebase on the current master).

The simple test is:
```python
from photutils import Background2D, BkgIDWInterpolator
data = np.arange(12).reshape(3, 4)
interpolator = BkgIDWInterpolator()
bkg = Background2D(data, (1, 1), filter_size=1,
                   interpolator=interpolator)
bkg.background
```
The result should the same as the input `data`.